SwanProjects/src/ProjectDialog.tsx

```diff
@@ -0,0 +1,190 @@
+// Copyright (c) SWAN Team.
+// Author: Omar Zapata CERN 2021
+
+/**
+ * ProjectDialog is a modal dialog that allows to create and edit projects,
+ * the dialog allows to select the stack, release, platform and write in a textbox
+ * a bash script to run inside the project.
+ *
+ * If the .swanproject file is corrupted for any reason, this dialog will appear
+ * for the user in order to recover the project with the right information provide by he/she.
```

Comment:
If the .swanproject file is corrupted, this dialog will appear for the user to enter the right information about the project again.